tests/integration/list/endpoints/test_endpoints_command.py

```diff
@@ -132,3 +132,32 @@ def test_stack_does_not_exist(self):
         self.assertIn(
             expected_output, command_result.stderr.decode(), "Should have raised error that outputs do not exist"
         )
+
+    def test_has_stack_name_table_output(self):
+        template_path = self.list_test_data_path.joinpath("test_endpoints_template.yaml")
+        stack_name = method_to_stack_name(self.id())
+        config_file_name = stack_name + ".toml"
+        region = boto3.Session().region_name
+        deploy_command_list = self.get_deploy_command_list(
+            template_file=template_path,
+            guided=True,
+            config_file=config_file_name,
+            region=region,
+            confirm_changeset=True,
+            disable_rollback=True,
+        )
+        deploy_process_execute = run_command_with_input(
+            deploy_command_list, "{}\n{}\nY\nY\nY\nY\nY\nY\n\n\nY\n".format(stack_name, region).encode()
+        )
+        cmdlist = self.get_endpoints_command_list(stack_name=stack_name, region=region, template_file=template_path)
+        command_result = run_command(cmdlist, cwd=self.working_dir)
+        header_string = "-+(-|\n)+R\n*e\n*s\n*o\n*u\n*r\n*c\n*e\n*( |\n)*I\n*D\n*( |\n)*P\n*h\n*y\n*s\n*i\n*c\n*a\n*l\n*( |\n)*I\n*D\n*( |\n)*C\n*l\n*o\n*u\n*d\n*( |\n)*E\n*n\n*d\n*p\n*o\n*i\n*n\n*t\n*s\n*( |\n)*M\n*e\n*t\n*h\n*o\n*d\n*s\n*( |\n)*-+(-|\n)+"
+        expression_list = [
```

Comment:
My main concern is that maintaining this going forward would be tricky. Someone coming in and changing the table structure would have to spend quite some time understanding these patterns to update the tests. Is there any way of simplifying or making these more readable?

Reply:
Added some explanation and the functions to generate them